programs/survival/bucketstack.sc

```diff
@@ -0,0 +1,133 @@
+__config() ->
+(
+	//change these number to whatever you want the maximum stack size to be
+	//note that the max scarpet can inventory_set() is 2^31 - 1
+	global_buckets = m(
+		l('water',1),
+		l('lava',1),
+		l('milk',1),
+		l('pufferfish',1),
+		l('salmon',1),
+		l('cod',1),
+		l('tropical_fish',1),
+		l('axolotl',1)
+	);
+	return(
+		m(
+			l('scope','player'),
+			l('stay_loaded',true)
+		)
+	);
+);
+
+__on_player_collides_with_entity(player,entity) ->
+(
+	if(entity~'pickup_delay' == 0,
+		l(item,count,nbt) = entity~'item';
+		if(item~'_bucket',
+			//if it's an item we're picking up that is a non-empty bucket
+			//first determine the max stack size
+			type = null;
+			while(type == null,length(global_buckets),
+				type = item~(keys(global_buckets):_);
+			);
+			max_stack = global_buckets:type;

```

Comment:
why not `max_stack = global_buckets:item;`? If you want to avoid conflicts with modded bucket items, you can set a default stack size like so:
`max_stack = global_buckets:item || 1;`

Maybe I'm just misunderstanding wht this does.

Reply:
Fixed. Thanks